src/tsutil/Regex.cc

```diff
@@ -120,6 +120,10 @@ RegexMatches::RegexMatches(uint32_t size)
     &RegexMatches::malloc, [](void *, void *) -> void {}, static_cast<void *>(this));
 
   _match_data = pcre2_match_data_create(size, ctx);
+  if (_match_data == nullptr) {
```

Comment:
Since the idea for having built-in storage is to prevent heap allocations when the passed in `size` is <= `DEFAULT_SIZE`, can we debug assert that `!(_match_data == nullptr && size == DEFAULT_SIZE)`?  This will alert devs when the expected size increases and this would allocate for the default.

Reply:
Yes, I think having a debug assert is a good idea - I will update the PR to include that